src/app/api/chat/route.ts

```
@@ -0,0 +1,304 @@
+
+// ============================================================
+// ROUTE HANDLER
+// ============================================================
+
+export async function POST(req: Request) {
+  try {
+    const { messages } = (await req.json()) as { messages: UIMessage[] };
+
+    // Extract latest user query from AI SDK v6 UIMessage parts
+    const lastMessage = messages[messages.length - 1];
+    
```

ESLint

[error] Delete `····`

(prettier/prettier)